docs/building-spark.md

```diff
@@ -16,24 +16,27 @@ Building Spark using Maven requires Maven 3.3.9 or newer and Java 7+.
 
 ### Setting up Maven's Memory Usage
 
-You'll need to configure Maven to use more memory than usual by setting `MAVEN_OPTS`. We recommend the following settings:
+If you are compiling with Java 7, you'll need to configure Maven to use more memory than usual by setting `MAVEN_OPTS`:
```

Comment:
I actually think the ReservedCodeCacheSize advice is defunct. I don't use it and don't have problems. But, maybe not worth worrying about here. It has nothing to do with the warnings discussed in this section though.

Anyway, might I finally suggest that, because Java 8 is more the default now and soon to be required, that this advice basically say:

Add Xmx and XX:ReservedCodeCacheSize to MAVEN_OPTS
Or else you might see this error
If you're on Java 7 by the way, also add this MaxPermSize param like so